packages/cdktf/lib/tfExpression.ts

```diff
@@ -374,6 +374,72 @@ export function call(name: string, args: Expression[]) {
   return new FunctionCall(name, args) as IResolvable;
 }
 
+export const FOR_EXPRESSION_KEY = ref("key");
+export const FOR_EXPRESSION_VALUE = ref("val");
+
+/**
+ * https://www.terraform.io/docs/language/expressions/for.html
+ */
+class ForExpression extends TFExpression {
+  constructor(
+    private input: Expression,
+    private valueExpression: Expression,
+    private keyExpression?: Expression
+  ) {
+    super({ input, valueExpression, keyExpression });
+  }
+
+  public resolve(context: IResolveContext): string {
+    const suppressBraces = context.suppressBraces;
+    context.suppressBraces = true;
+    const key = this.resolveArg(context, FOR_EXPRESSION_KEY);
+    const value = this.resolveArg(context, FOR_EXPRESSION_VALUE);
+    const input = this.resolveArg(context, this.input);
+    const valueExpr = this.resolveArg(context, this.valueExpression);
+    const keyExpr = this.keyExpression
+      ? this.resolveArg(context, this.keyExpression)
+      : undefined;
+
+    let expr: string;
+    if (keyExpr) {
+      expr = `{ for ${key}, ${value} in ${input}: ${keyExpr} => ${valueExpr} }`;
+    } else {
+      expr = `[ for ${key}, ${value} in ${input}: ${valueExpr}]`;
+    }
+
+    return suppressBraces ? expr : `\${${expr}}`;
+  }
+}
+
+/**
+ * Creates a for expression, used for advanced internal use cases (e.g. in the @cdktf/aws-adapter)
+ * It is currently not possible to use this via JSII in other languages. Usually you might want to
+ * use TerraformIterator instead.
+ *
+ * @param input the input for the expression e.g. a list or an object
+ * @param valueExpression will express the value e.g. [ for i in &lt;input&gt; : &lt;valueExpression&gt; ]
+ * @param keyExpression if set will return an object e.g. { for k,v in &lt;input&gt; : &lt;keyExpression&gt; => &lt;valueExpression&gt;}
```

Comment:
Is the `&lt;` and `&gt;` by design or accident?

Reply:
by design, if you hover over the method name in e.g. VSCode you can see that they look right this way (else they were treated as html tags)